src/NetMPA.Catalog/NetMPA.Catalog.Api/Controllers/CatalogController.cs

```diff
@@ -0,0 +1,216 @@
+
+        /// <summary>
+        /// Category by Id
+        /// </summary>
+        [HttpGet()]
+        [Route("category/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<Category>> GetCategoryById(int id)
+        {
+            var result = await this.categoryService.Get(id);
+            if (result == null) return NotFound();
+            return Ok(this.mapper.Map<Category>(result));
+        }
+
+        /// <summary>
+        /// Add Category
+        /// </summary>
+        [HttpPost]
+        [Route("category")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult> AddCategory([FromBody] CategoryAdd category)
+        {
+            if (category == null) return BadRequest();
+
+            await this.categoryService.Add(new Bll.Models.Category()
+                {
+                    Name = category.Name,
+                    Image = category.Image,
+                    Parent = new Bll.Models.Category() { Id = category.ParentId }
+                }
+            );
+
+            return StatusCode(201);
+        }
+
+        /// <summary>
+        /// Update Category
+        /// </summary>
+        [HttpPut]
+        [Route("category/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult> UpdateCategory([FromRoute] int categoryId, [FromBody] CategoryAdd category)
+        {
+            if (category == null) return BadRequest();
+
+            await this.categoryService.Add(new Bll.Models.Category()
+            {
+                Name = category.Name,
+                Image = category.Image,
+                Parent = new Bll.Models.Category() { Id = category.ParentId }
+            }
+            );
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// Delete Category
+        /// </summary>        
+        [HttpDelete]
+        [Route("category/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult> DeleteCategory([FromRoute] int categoryId)
+        {
+            if (categoryId <= 0) return BadRequest();
+
+            await this.categoryService.Delete(categoryId);
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// List of Products
+        /// </summary>
+        [HttpGet()]
+        [Route("product")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<IEnumerable<Product>>> GetAllProducts()
+        {
+            var result = await this.productService.List();
+            if (result == null) return NotFound();
```

Comment:
It's better to return empty list and status code 200. 404 is for the case when a specific product is not found by id

Reply:
Requirement:
`List of Items (filtration by category id and pagination)`
is not implemented

Reply:
implemented